pkg/storage/cacher.go

```diff
@@ -307,21 +381,53 @@ func (c *Cacher) Codec() runtime.Codec {
 	return c.storage.Codec()
 }
 
+func (c *Cacher) indexValues(event *watchCacheEvent) ([]string, bool) {
+	if c.indexFunc == nil {
+		return nil, false
+	}
+	// TODO: Currently we assume that only one index exists, thus
```

Comment:
We can't do it. What we assume is that "indexFunc" will return 0 or 1 value. But there can be a different between previous and current object so we need to return more than 1 value.